neo/rawio/biocamrawio.py

```diff
@@ -0,0 +1,203 @@
+        self._mea_pitch = mea_pitch
+        self._verbose = verbose
+
+    def _source_name(self):
+        return self.filename
+
+    def _parse_header(self):
+        try:
+            import h5py
+            HAVE_H5PY = True
+        except ImportError:
+            HAVE_H5PY = False
+        assert HAVE_H5PY, 'h5py is not installed'
+        self._rf, self._num_frames, self._sampling_rate, self._num_chanels, self._ch_indices, \
+        self._file_format, self._signal_inv, self._positions, self._read_function = openBiocamFile(
+            self.filename, self._mea_pitch, self._verbose)
+
+        signal_streams = np.array([('Signals', '0')], dtype=_signal_stream_dtype)
+
+        sig_channels = []
+        for c in range(self._num_chanels):
+            ch_name = 'ch{}'.format(c)
+            chan_id = str(c + 1)
+            sr = self._sampling_rate  # Hz
+            dtype = "uint16"
+            units = 'uV'
+            gain = 1.  # TODO find gain
+            offset = 0.  # TODO find offset
+            stream_id = '0'
+            sig_channels.append((ch_name, chan_id, sr, dtype, units, gain, offset, stream_id))
+        sig_channels = np.array(sig_channels, dtype=_signal_channel_dtype)
+
+        # No events
+        event_channels = []
+        event_channels = np.array(event_channels, dtype=_event_channel_dtype)
+
+        # No spikes
+        unit_channels = []
+        unit_channels = np.array(unit_channels, dtype=_spike_channel_dtype)
+
+        self.header = {}
+        self.header['nb_block'] = 1
+        self.header['nb_segment'] = [1]
+        self.header['signal_streams'] = signal_streams
+        self.header['signal_channels'] = sig_channels
+        self.header['unit_channels'] = unit_channels
+        self.header['event_channels'] = event_channels
+
+        self._generate_minimal_annotations()
+
+    def _segment_t_start(self, block_index, seg_index):
+        all_starts = [[0.]]
+        return all_starts[block_index][seg_index]
+
+    def _segment_t_stop(self, block_index, seg_index):
+        t_stop = self._num_frames / self._sampling_rate
+        all_stops = [[t_stop]]
+        return all_stops[block_index][seg_index]
+
+    def _get_signal_size(self, block_index, seg_index, stream_index):
+        assert stream_index == 0
+        return self._num_frames
+
+    def _get_signal_t_start(self, block_index, seg_index, stream_index):
+        assert stream_index == 0
+        return self._segment_t_start(block_index, seg_index)
+
+    def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop,
+                                stream_index, channel_indexes):
+        if i_start is None:
+            i_start = 0
+        if i_stop is None:
+            i_stop = self._num_frames
+
+        data = self._read_function(self._rf, i_start, i_stop, self._num_chanels)
+        # transform to slice if possible
+        if sorted(channel_indexes) == channel_indexes and np.all(np.diff(channel_indexes) == 1):
```

Comment:
In neo we already have this mechanism in base class here https://github.com/NeuralEnsemble/python-neo/blob/master/neo/rawio/baserawio.py#L514
So please remove this logic in the rawio.

Reply:
removed